### PR TITLE
Added support for environment variable substitution in config

### PR DESCRIPTION
**Pull request by emulbreh**
Syntax borrowed from [Circus](http://circus.readthedocs.org/en/latest/)

### CODE REVIEW

File: lymph/config.py [hunk 1]
```diff
@@ -112,8 +113,9 @@ def __repr__(self):
 
 
 class Configuration(ConfigObject):
-    def __init__(self, values=None):
+    def __init__(self, values=None, env=os.environ):
```
Comment (eguven) [inline lymph/config.py]: mutable data type as kwarg default?

Comment (emulbreh) [inline lymph/config.py]: os.environ is a global singleton, we explicitly want it to be the same object for every call.

File: docs/configuration.rst [hunk 2]
```diff
@@ -21,6 +21,17 @@ Configuration
...
 
 
+Environment Variables
+~~~~~~~~~~~~~~~~~~~~~~
+
+Lymph config files support environment variable substitution for string values:
+
+.. code-block:: yaml
+
+    key: protocol://$(env.USER):$(env.PASSWORD)@host/path
```
Comment (mouadino) [inline docs/configuration.rst]: why env.<> ? why not use python string substitution ?

     key: protocol://{USER}:{PASSWORD}@host/path

And then we just do before parsing yaml:

    config = config.format(**os.environ)

Comment (emulbreh) [inline docs/configuration.rst]: string substitution is too useful as config values, e.g. 
```
greeting: "Hello {name}"
```

Comment (mouadino) [inline docs/configuration.rst]: Then how about using ``%(USER)s`` syntax ?

Comment (emulbreh) [inline docs/configuration.rst]: "string substitution is too useful in config values".

The ``$()`` syntax is extensible (might be useful to unify dependencies, load config from zk, ..) and copies a feature of circus.

**Comment by eguven** [comment]
+1

**Pull request merged**